### ISSUE TITLE
Filter "Your sites" by tags.

### ISSUE THREAD

**Issue by twMat** [simonbaird/tiddlyhost#75]
So I have *only* seven wikis but it is already obvious that very soon the Your sites list will be difficult to overview. I wish for something like in the image. This will also encourage people to actually tag their wikis which whill make public ones more easily findable:

![image](https://user-images.githubusercontent.com/4534056/109984954-8d5dad80-7d04-11eb-9f55-382182f169dd.png)

**Comment by simonbaird** [comment]
+1 Nice idea.

**Comment by simonbaird** [comment]
It's not quite what you wanted here, but there's now a search box on the "Your sites" page, and it matches tags. So you can search for a tag name and see sites with that tag.

**Comment by twMat** [comment]
OK... may I offer some blunt criticism of it?

**Comment by simonbaird** [comment]
Of course! It would be much appreciated. :grin:

**Comment by twMat** [comment]
OK, that search feature does not provide anything that simply Ctrl-F (win) doesn't. Even a tad worse, since one has to deliberately click the button after wheras Ctrl-F searches live once it is active. That also goes for searching for tags as the are eplicit on the page and Ctrl-F makes no distinction what the string searched for "is".

So, the featured search would be more useful if it allowed for live search - i.e the results are gradually and visually filtered down as one is typing. Also distinguish the results as e.g "wiki title", "tag" perhaps also "description"

Second, its position and design is (probably) very easy to miss but, admittedly, this may be because my own extensive list of wikis may have primed me to immediately shift my focus to there.

...but on the other hand, anyone who only has a few wikis won't have any use for the search feature anyway. - Actually, what's the average number of wikis for active users anyway? I have understood that I'm on the extreme end.

The two aspects above - i.e what/how it actually searches, and who is to use such a search feature - makes me think that my OP about showing *tags* makes more sense and, again, it would foster good (personal) tag usage which then benefits everyone when wikis are made public. (My assumption here is that tags are the main "identifier" used when people locate stuff among hub listed wikis, including the new "template" ideas.)

...but back to the position/layout:

One idea (which I'm not sure is very good) might be to emulate the very position and look of TW own search field, because people are familiar with that.

But I'd think a more useful display would simply be more prominent. Here's a crude illustration to show the idea (where I've also moved the statistics for "number of sites" and GB.

<img width="632" alt="image" src="https://github.com/user-attachments/assets/e2955606-12fd-4375-886e-add58dd974f0">

....

Further, I get this annoyance when entering the search field. The shown items are really searches from my *github* search field. I suspect it comes from using a very generic field name or some such. (Similar things have happened for me with other services):

<img width="311" alt="image" src="https://github.com/user-attachments/assets/157978aa-d748-43fc-bbb9-d265338900c5">

...

OK, hope some of this makes sense :-)

**Comment by twMat** [comment]
Indeed, I'd say it is `name="q"`, which is identical to github issues search field, causing it 

<img width="478" alt="image" src="https://github.com/user-attachments/assets/bb766e90-55e0-474e-a320-2fa0bed610f3">

**Comment by simonbaird** [comment]
Thanks for the suggestions. I think moving the search box to the left is a good idea.

Having it search while you type is also reasonable, I'll take a look at that some time.

I'm not sure about changing the field name to avoid your GitHub search autocomplete - it seems like a browser configuration thing maybe...?

**Comment by twMat** [comment]
Hehe, OK, I spent a few confused minutes before getting it! As you can guess I think it'll need something more added to it to be "findable". Do you think the UI in the OP is not good?

While at this, I think it is worth reflecting on the broader role of the tags in TH as it may affect the UI also for the issue. I think a main "value add" with the tags is for making public+searchable wikis findable. So it is good if users are incentivised to tag their wikis as "good as possible", perhaps even as *much* as possible? I think I've outlined some ideas around this elsewhere but can elaborate if you want me to. Anyway, my main point is, as noted, that maybe there are aspects now that can synergize with the broader role of tags in TH.

By the way, it is a very nice pedagogical touch that the tags are styled the same as in TW itself :-D
